generic/tclClock.c

```diff
@@ -3472,6 +3474,21 @@ ClockParseFmtScnArgs(
     /* check base fields already cached (by TZ, last-second cache) */
     if ( dataPtr->lastBase.timezoneObj == opts->timezoneObj
       && dataPtr->lastBase.date.seconds == baseVal) {
+	canUseLastBase = 1;
+	/*
+	 * There's one exception: If TZ changed, and if timezone is just the
+	 * :localtime placeholder, then it's not safe to re-use the cached
+	 * base time.
+	 */
+	if ( oldTZEpoch != dataPtr->lastTZEpoch
+	  && opts->timezoneObj->length == dataPtr->literals[LIT_LOCALTIME]->length
```

Comment:
*If not removed due to suggestion above (compare offsets or unneeded)...*
This is not safe, if for some reason the normalization process (in `ClockSetupTimeZone`) would not update string representation of `opts->timezoneObj`, this could segfault theoretically. 
Fully correct would be something like this:
```diff
- 	  && opts->timezoneObj->length == dataPtr->literals[LIT_LOCALTIME]->length
-	  && !memcmp(opts->timezoneObj->bytes, dataPtr->literals[LIT_LOCALTIME]->bytes, opts->timezoneObj->length) 
+	  && strcmp(TclGetString(opts->timezoneObj), Literals[LIT_LOCALTIME]) == 0
 ```
But... if `NormTimezoneObj` would be extended to consider `:localtime` literal (properly normalize it to a single object everytime), it would be faster and this could be just:
```diff
- 	  && opts->timezoneObj->length == dataPtr->literals[LIT_LOCALTIME]->length
-	  && !memcmp(opts->timezoneObj->bytes, dataPtr->literals[LIT_LOCALTIME]->bytes, opts->timezoneObj->length) 
+	  && opts->timezoneObj == dataPtr->literals[LIT_LOCALTIME]
 ```

Reply:
Good catch. That's substantially more readable, too. For now, I'll open a PR with this suggestion.